src/weak.rs

```diff
@@ -0,0 +1,93 @@
+use ::*;
```

Comment:
this is non-idiomatic, please use only types that are needed
also, how does it compile given no module documentation?